pennylane/templates/layers.py

```diff
@@ -0,0 +1,418 @@
+sequence in each call.
+
+
+Qubit architectures
+-------------------
+
+Strongly entangling circuit
+***************************
+
+.. autosummary::
+
+    StronglyEntanglingLayers
+    StronglyEntanglingLayer
+
+Random circuit
+**************
+
+.. autosummary::
+
+    RandomLayers
+    RandomLayer
+
+Continuous-variable architectures
+---------------------------------
+
+Continuous-variable quantum neural network
+******************************************
+
+.. autosummary::
+
+    CVNeuralNetLayers
+    CVNeuralNetLayer
+
+Interferometer
+**************
+
+.. autosummary::
+
+    Interferometer
+
+Code details
+^^^^^^^^^^^^
+"""
+#pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from collections.abc import Sequence
+
+from pennylane.ops import CNOT, RX, RY, RZ, Rot, Squeezing, Displacement, Kerr, Beamsplitter, Rotation
+from pennylane.qnode import QuantumFunctionError
+from pennylane.variable import Variable
+import numpy as np
+
+
+def StronglyEntanglingLayers(weights, ranges=None, imprimitive=CNOT, wires=None):
+    """A sequence of layers of type :func:`StronglyEntanglingLayer()`, as specified in :cite:`schuld2018circuit`.
+
+    The number of layers :math:`L` is determined by the first dimension of ``weights``. The template is applied to
+    the qubits specified by the sequence ``wires``.
+
+    Args:
+        weights (array[float]): array of weights of shape ``(L, len(wires), 3)``
+
+    Keyword Args:
+        ranges (Sequence[int]): sequence of ranges determining the architecture of the imprimitive gates in each layer
+        imprimitive (pennylane.ops.Operation): two-qubit gate to use, defaults to :class:`~.CNOT`
+        wires (Sequence[int]): sequence of qubit indices that the template acts on
+    """
+
+    if ranges is None:
+        ranges = [1]*len(weights)
+
+    for block_weights, block_range in zip(weights, ranges):
+        StronglyEntanglingLayer(block_weights, r=block_range, imprimitive=imprimitive, wires=wires)
+
+
+def StronglyEntanglingLayer(weights, r=1, imprimitive=CNOT, wires=None):
+    """A layer applying rotations on each qubit followed by cascades of 2-qubit entangling gates.
+
+    The 2-qubit or imprimitive gates act on each qubit :math:`i` chronologically. The second qubit for
+    each gate is determined by :math:`(i+r)\mod n`, where :math:`n` is the number of qubits
+    and :math:`r` a layer hyperparameter called the range.
+
+    This is an example of two 4-qubit strongly entangling layers (ranges :math:`r=1` and :math:`r=2`) with
+    rotations :math:`R` and CNOTs as imprimitives:
+
+    .. figure:: ../../_static/layer_sec.png
+        :align: center
+        :width: 60%
+        :target: javascript:void(0);
+
+    Args:
+        weights (array[float]): array of weights of shape ``(len(wires), 3)``
+
+    Keyword Args:
+        r (Sequence[int]): range of the imprimitive gates of this layer
```

Comment:
um, for a single layer r can not be sequence[init]

Reply:
Good catch!